src/osd/ReplicatedPG.cc

```diff
@@ -1310,6 +1315,54 @@ void ReplicatedPG::do_pg_op(OpRequestRef op)
   delete filter;
 }
 
+int ReplicatedPG::do_scrub_ls(MOSDOp *m, OSDOp *osd_op)
+{
+  if (!pool.info.supports_omap()) {
+    return -EINVAL;
+  }
+  if (m->get_pg() != info.pgid.pgid) {
+    dout(10) << " scrubls pg=" << m->get_pg() << " != " << info.pgid << dendl;
+    return -EINVAL; // hmm?
+  }
+  auto bp = osd_op->indata.begin();
+  ObjectOperation::scrub_ls_arg_t arg;
+  try {
+    arg.decode(bp);
+  } catch (buffer::error&) {
+    dout(10) << " corrupted scrub_ls_arg_t" << dendl;
+    return -EINVAL;
+  }
+  ::encode(info.history.same_interval_since, osd_op->outdata);
```

Comment:
I think we probably want to encode a wrapper object here so that we can use the ENCODE_START macros.  Probably in the same common/ file where the encode wrappers are?